src/coord/src/coord.rs

```diff
@@ -2106,14 +2106,40 @@ impl Coordinator {
 
     async fn sequence_create_secret(
         &mut self,
+        session: &Session,
         plan: CreateSecretPlan,
     ) -> Result<ExecuteResponse, CoordError> {
         let CreateSecretPlan {
             name,
-            secret: _,
+            secret,
             if_not_exists,
         } = plan;
 
+        let compute_instance = self
+            .catalog
+            .resolve_compute_instance(session.vars().cluster())?
+            .id;
+
+        let temp_storage = RowArena::new();
+        self.dataflow_builder(compute_instance).prep_scalar_expr(
+            &mut secret.secret_as.clone(),
+            ExprPrepStyle::OneShot {
+                logical_time: None,
+                session,
+            },
+        )?;
```

Comment:
Oh, funny, we definitely don't want to be requiring a compute instance here, but yeah, `prep_scalar_expr` is currently impossible to call without one. I think it should all work out if you pop `DataflowBuilder::{prep_relation_expr, prep_scalar_expr, eval_unmaterializable_func}` out into free functions that take a `CatalogState` as an argument, rather than as methods on `DataflowBuilder`. Then you'll be able to call `prep_scalar_expr` without resolving a compute instance.